Mathlib/Topology/ContinuousOn.lean

```diff
@@ -978,6 +978,13 @@ theorem Continuous.comp_continuousOn {g : β → γ} {f : α → β} {s : Set α
   hg.continuousOn.comp hf (mapsTo_univ _ _)
 #align continuous.comp_continuous_on Continuous.comp_continuousOn
 
+@[fun_prop]
+theorem Continuous.comp_continuousOn'
```

Comment:
I just realized that `fun_prop` stores only one composition theorem for each function property. This is true for any lambda theorem (_id,_const,_comp,_pi,_apply). Thus this this theorem clashes with `ContinuousOn.comp''` and `ContinuousOn.comp'`.

The current behavior is that only the latest lambda theorem is used. I think it is ok to keep it marked as `fun_prop` for now. I will add a support for multiple variants of lambda theorems, I already have this working for `fun_trans` so it should be quite easy.